src/components/Registration/index.jsx

```diff
@@ -0,0 +1,64 @@
+import React,{useState} from 'react'
+import userIco from './../Registration/user_icon.png';
+
+const Registration = () => {
+
+  const [user,setUser] = useState(
+    {
+      username: '',
+      email: '',
+      password: '',
+      passwordConfirm: '',
+    }
+ )
+
+ const handleSubmit = (e) => {
+  e.preventDefault() 
+  if (user.password === user.passwordConfirm && user.username !== '' && user.email !== '' && user.password !== '' && user.passwordConfirm !== '') {
+    console.log(user)
+  } else if (user.password !== user.passwordConfirm) {
+    alert('Hesla se neshodují')
+  } else {
+    alert('Nemáte vyplněná všechna povinná pole')
+  }
+ }
+
+ const handleChangeEmail = (e) => {
+  if (user.username === '' && e.target.value.includes('@')) {
+    const nameToUsername = e.target.value.split('@') 
+    setUser({...user, username: nameToUsername[0], email: e.target.value})
+  } else {
+    setUser({...user, email: e.target.value})
+  }
```

Comment:
Tohle je dobrá myšlenka! Akorát bych to dal spíš na nějaký field blur (tj když uživatel z políčka odejde, nebo aspoň chvíli nepíše) než na každou změnu - to pak se uživateli hýbe moc věcí na obrazovce když píše (=blbý z UX hlediska protože mu to rozptyluje pozornost)

Reply:
A až teď mi došlo že to nebude fungovat dobře :D 
`if (user.username === '' && e.target.value.includes('@')) {`
tohle bude fungovat jen jednou, tj pokud uživatel napíše email postupně (ve chvíli kdy tam napíše zavináč). Ale pokud udělá chybu a bude ho chtít upravit, už se mu to nezmění